modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -66,6 +90,23 @@ abstract class HealthCheckConsensus[
       if (inProgress.nonEmpty) manageRounds(inProgress) else Applicative[F].unit
     }
 
+  private def manageMissingProposals(
+    missingFromPeers: Set[PeerId],
+    key: K,
+    round: HealthCheckConsensusRound[F, K, A, B, C]
+  ): F[Unit] =
+    round.getRoundIds.flatMap { roundIds =>
+      Applicative[F].whenA(missingFromPeers.size > 0) {
+        logger.debug(
+          s"Missing proposals for round ids: ${roundIds} for key: ${key} are from peers: ${missingFromPeers}. Requesting proposals"
+        )
+      }
+    } >>
+      missingFromPeers.toList.traverse { requestProposal(_, round.getOwnRoundId) }
```

Comment:
I'm not sure if asking for the proposal based only on your own `roundId` instead of basing it on all merged ids is correct. If the peer doesn't know your rounds id yet it doesn't mean that it doesn't know other ids that this whole round globally consists of. Not sure, but I think that's how worked in constellation.